pkg/services/ngalert/provisioning/persist.go

```diff
@@ -24,3 +24,20 @@ type ProvisioningStore interface {
 type TransactionManager interface {
 	InTransaction(ctx context.Context, work func(ctx context.Context) error) error
 }
+
+type ProvenanceOrgAdapter struct {
+	Inner models.ProvisionableInOrg
+	OrgID int64
+}
+
+func (a ProvenanceOrgAdapter) ResourceType() string {
```

Comment:
Do you want `a *ProvenanceOrgAdapter`? I'm neutral on the question. How long-lived are these structs?

Reply:
Very short lived. They're just simply mapping one interface to another in a few one-off locations.